fake-lmp-device-register

```diff
@@ -23,46 +28,77 @@ class Options(NamedTuple):
     production: bool
 
 
-def create_key(uuid: str, factory: str, production: bool) -> Tuple[bytes, bytes]:
-    key = subprocess.check_output(
-        ["openssl", "ecparam", "-genkey", "-name", "prime256v1"]
-    )
+def create_key(uuid: str, factory: str, production: bool, sota_dir: str) -> Tuple[bytes, bytes]:
+    # Set path for the private key file
+    keyfile_path = os.path.join(sota_dir, "pkey.pem")
+
+    try:
+        log.info("Creating key for UUID: %s, Factory: %s", uuid, factory)
+
+        # Generate a private key into current-dir/var/sota
+        subprocess.check_call(["openssl", "ecparam", "-genkey", "-name", "prime256v1", "-out", keyfile_path])
+
+        # Read the key from the file
+        with open(keyfile_path, "rb") as keyfile:
+            key_data = keyfile.read()
+
+        # Create a temp config for OpenSSL's CSR generation
+        with NamedTemporaryFile(delete=False) as cnf:
+            cnf.write(f"""[req]
+    prompt = no
+    distinguished_name = dn
+    req_extensions = ext
+
+    [dn]
+    CN={uuid}
+    OU={factory}
+    """.encode()
+            )
+            if production:
+                cnf.write(b"businessCategory=production\n")
+            cnf.write(
+                b"""
+    [ext]
+    keyUsage=critical, digitalSignature
+    extendedKeyUsage=critical, clientAuth
+    """
+            )
+            cnf.flush()
+            cnf.close()
+
+            # Use OpenSSL to generate a CSR using the private key and config
+            r = subprocess.run(["openssl", "req", "-new", "-config", cnf.name, "-key", keyfile_path], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            if r.returncode != 0:
+                log.error("Error in running openssl command:")
+                log.error(r.stderr.decode())
+
+                # Clean up temp files in case of errors
+                os.unlink(cnf.name)
+                raise subprocess.CalledProcessError(r.returncode, r.args)
+
+    except Exception as e:
+        log.error("An error occurred: %s", e)
+        raise
+    finally:
+        # Let's ensure that temp files are deleted
+        os.unlink(cnf.name)
```

Comment:
you don't have to do these unlink calls. let the context delete the tempfile by removing `delete=False`, and remove the call to `cnf.close()` as the `cnf.flush` is enough to get it to disk.